mnist/testing/predict_test.py

```diff
@@ -0,0 +1,91 @@
+"""Test mnist_client.
+
+This file tests that we can send predictions to the model
+using REST.
+
+It is an integration test as it depends on having access to
+a deployed model.
+
+We use the pytest framework because
+  1. It can output results in junit format for prow/gubernator
+  2. It has good support for configuring tests using command line arguments
+     (https://docs.pytest.org/en/latest/example/simple.html)
+
+Python Path Requirements:
+  kubeflow/testing/py - https://github.com/kubeflow/testing/tree/master/py
+     * Provides utilities for testing
+
+Manually running the test
+ 1. Configure your KUBECONFIG file to point to the desired cluster
+"""
+
+import json
+import logging
+import os
+import requests
+import six
+
+from kubernetes.config import kube_config
+from kubernetes import client as k8s_client
+
+import pytest
+
+from kubeflow.testing import util
+
+def test_predict(master, namespace):
+  app_credentials = os.getenv("GOOGLE_APPLICATION_CREDENTIALS")
```

Comment:
Is it better to encapsulate line 36-59 as a function in `kubeflow.testing.util` for reuse?

Reply:
Once the code stabilizes probably. The problem right now is that putting it in kubeflow.testing means splitting it across two different repos and thus two different PRs. This is pretty painful during the early stages when we detect bugs and need to fix them.

e.g some of the PY3 issues only cropped up on the test system because I use PY2.7 locally. Similarly, the gcloud service account code is specific to running on the test cluster rather than locally.